crates/ruff_linter/src/rules/flake8_pyi/rules/string_or_bytes_too_long.rs

```diff
@@ -59,6 +59,10 @@ pub(crate) fn string_or_bytes_too_long(checker: &mut Checker, string: StringLike
         return;
     }
 
+    if semantic.in_annotation() {
```

Comment:
I wondered if it might be better to be more targeted in our exclusions here, since there are relatively few valid annotations in stub files that would involve long string literals -- basically just `Literal` and `Annotated`, I think? We can detect if we're in a `typing.Literal` slice like this:

```suggestion
    if semantic.in_typing_literal() {
```

But we don't currently track in the semantic model whether we're in an `Annotated` slice in the same way, and I'm not sure if it's worth it to add that tracking just for this, as I'm not sure it would practically get us much.

TL;DR -- this seems good; thanks!